File: in_Java/0695 Max Area of Island.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int maxres = Integer.MIN_VALUE;
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                if(grid[i][j] == 1){
                    int cur = dfs(grid,i,j);
                    maxres = Math.max(cur,maxres);
                }
            }
        }
        return maxres == Integer.MIN_VALUE? 0 : maxres;
    }
    private int dfs(int[][] g, int i, int j){
        int n = 0;
        if(i < 0 || j < 0 || i >= g.length || j >= g[0].length || g[i][j] != 1) return 0;
        g[i][j] = -1;
        n++;
        n += dfs(g,i-1,j);
        n += dfs(g,i+1,j);
        n += dfs(g,i,j-1);
        n += dfs(g,i,j+1);
        return n;
    }
}

File: in_Python/0695 Max Area of Island.py
class Solution(object):
    def maxAreaOfIsland(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        res = 0
        if grid == None:
            return res
        if len(grid) == 0 or len(grid[0]) == 0:
            return res
        directions = [[1,0],[-1,0],[0,1],[0,-1]]
        row = len(grid)
        col = len(grid[0])
        for i in range(row):
            for j in range(col):
                if grid[i][j] == 0 or grid[i][j] == -1:
                    continue
                q = []
                size = 0
                grid[i][j] = -1
                q.append([i,j])
                while len(q) != 0:
                    h = q.pop(0)
                    size += 1
                    thisi = h[0]
                    thisj = h[1]
                    for d in directions:
                        nexti = thisi + d[0]
                        nextj = thisj + d[1]
                        if 0 <= nexti < row and 0 <= nextj < col and grid[nexti][nextj] == 1:
                            q.append([nexti,nextj])
                            grid[nexti][nextj] = -1
                res = max(res,size)
        return res
                    